### ISSUE TITLE
Spotify client not found.

### ISSUE THREAD

**Issue by trjuntunen** [evilarceus/Spotify-PlayCount#2]
I followed the steps exactly in the README to in order to be able to pull play counts from Spotify (the publicly hosted API works, I'm simply trying to get it to work on my own machine). After completing every single step carefully, (including replacing .spa file, running zlink script ect.) and then running Spotify using the command given in the README (with the appropriate args e.g.  "--allow-running-insecure-content" ect.) I get a "Spotify client could not be found" error when trying to make a request to the endpoint which I have specified in the config.json (which looks like this btw):
`{
    "port": 8080,
    "useSecure": false,
    "secure": {
        "cert": "",
        "key": "",
        "chain": "'"
    },
    "http": {
        "endpoint": "/albums"
    },
    "websocket": {
        "ipAddress": "127.0.0.1"
    }
}`

I'm using index.js to start the simple node server so obviously https isn't necessary, so I left properties blank in "secure".

The URL I was trying to make a GET request to was: `http://127.0.0.1:8080/albums?albumid=49K82qcZN4u109e6Zw8io1`. I tried other forms of this url to no avail. 

Also, when I use the command `spotify --ignore-certificate-errors --allow-running-insecure-content` it loads the Spotify app normally but I immediately get a Mac notification saying "Session: Spotify (sleep) just terminated" and my terminal shuts down. 

Do you have any advice or tips in order to get this to work for myself on my own machine? Since your publicly hosted API works I imagine it may just be something I'm doing wrong. Let me know if more information is needed in order to get to a solution.

Thanks.

**Comment by entriphy** [comment]
Hi, can you send me your `zlink.spa` file that was generated from `spotify_zlink.js`?

**Comment by trjuntunen** [comment]
Thanks for the reply. I uploaded it to Google Drive with a shareable link here: https://drive.google.com/file/d/19KVyymbhB2H9eakQqW3hFrnSyGXz26mp/view

After running the spotify_zlink.js script it DID say it was successful also.

**Comment by entriphy** [comment]
I tried your zlink file on Mac and Linux and it works fine for me. Can you try launching Spotify without any arguments? It should still work without the extra arguments using your config.

Also, if you can, install [spicetify-cli](https://github.com/khanhas/spicetify-cli) and use it to enable the Chrome DevTools in Spotify (`spicetify enable-devtool`, it should then be available under the menu bar: Develop > Developer Tools). Then, screenshot the log under Console and post it here.

**Comment by trjuntunen** [comment]
I followed your steps. Here is the error in the Spotify developer console that appears on starting up Spotify:
`Uncaught (in promise) Error: Cosmos Request Error. Status: 502
    at zlink.bundle.js:160
    at zlink.bundle.js:57
    at zlink.bundle.js:17
    at zlink.bundle.js:17`

When I attempt to make a request to the endpoint, I get the message 
`{
"success": false,
"data": "An unknown error has occurred; logged to server console"
}`

And then on the console it prints an object like this:
`{uuid: <my spotify UUID>, id: "1TKCBDHIWLiqjWTxomqniD"}`

And at the same time my terminal window that's running the index.js server outputs the error 
`ReferenceError: print is not defined at WebSocket.listener (/Users/<username>/desktop/Spotify-PlayCount-master/app.js:29:13)`

So it seems I have made progress...index.js shows that my client is now connected, and I'm obviously not having the Spotify client not found error anymore. 

I used your command to open up Spotify with and without the arguments. Which Spotify CLI are you using to open up Spotify with the `spotify` command though? I'm using [Shpotify](https://github.com/hnarayanan/shpotify) I didn't see anything about a Spotify CLI in your README so maybe we're using different ones and this matters or something? 

Also just a sidenote, I tried using a different port (8000 instead of 8080) and I noticed that when I try a different port, I get an additional error in the Spotify developer console window saying `WebSocket connection to 'ws://127.0.0.1:8080/' failed: Error in connection establishment: net::ERR_CONNECTION_REFUSED`. (Notice it still says 8080 even though I changed config.json to be port 8000. When I run index.js it says it's running on port 8000 as well). Not sure if this is relevant to the issue just thought that was strange.

Let me know if you have any other suggestions to try! Or if any other information is needed. Thanks for the help so far.
